### PR TITLE
FE-60 FE-61 Save editors periodically and update overview on rename + delete

### PR DESCRIPTION
**Pull request by gidonkatten**
Save editors every 5 seconds. 
Save editor you came from when switching editors so you capture any last second changes
Save editor new name on rename
Delete editor cookie on delete

NOTE: Do not save on new editor as that will be done in at most 5 seconds time

Delete node in overview on editor delete
Rename node in overview on editor rename

TODO: FE-65 Set cookies after loading editors from file (currently only clear cookies)
TODO: Disable renaming of nodes in overview??????

### CODE REVIEW

File: src/components/buttons/DeleteEditorButton.vue [hunk 1]
```diff
@@ -1,20 +1,37 @@
 <template>
-  <span @click="deleteEditor({editorType, index})">
+  <span @click="this.deleteEditor">
```
Comment (tomproberts) [inline src/components/buttons/DeleteEditorButton.vue]: You don't need to write `this.deleteEditor`, you can just write `deleteEditor`.

File: src/components/buttons/RenameEditorButton.vue [hunk 2]
```diff
@@ -8,21 +8,42 @@
...
   @Prop({ required: true }) readonly editorType!: EditorType;
   @Prop({ required: true }) readonly index!: number;
+  @Prop({ required: true }) readonly oldName!: string;
   @Getter('editorNames') editorNames!: Set<string>;
+  @Getter('saveWithNames') saveWithNames!: SaveWithNames;
+  @Getter('editor') getEditor!: (editorType: EditorType, index: number) => EditorModel;
+  @Getter('overviewEditor') overviewEditor!: EditorModel;
   @Mutation('renameEditor') rename!:
-    (arg0: { editorType: EditorType; index: number; name: string}) => void;
+    (arg0: { editorType: EditorType; index: number; name: string; oldName: string}) => void;
```
Comment (tomproberts) [inline src/components/buttons/RenameEditorButton.vue]: I think there should be a space at the end of the object type, i.e. ...`oldName: string })`

**Review by lancelotblanchard — DISMISSED** [review]
Looks great! 👏

**Review by tomproberts — COMMENTED** [review]
Good stuff! Just 2 very minor things because I've decided to be extremely pedantic, sorry haha.
Either merge in (because Lancelot approved it) or change the 2 things first and I'll approve it, I don't mind too much.

I assume it's a lot easier than doing the debounce stuff? I suppose it seems a tiny bit odd that even if you literally change nothing, `updateNodeInOverview` is called every 5 seconds.

**Comment by gidonkatten** [comment]
> I suppose it seems a tiny bit odd that even if you literally change nothing, `updateNodeInOverview` is called every 5 seconds.

We can see if it becomes a problem ie we get some sort of lag and change the intervals accordingly. This is the downside of autosaving at fixed intervals (we also save unnecessarily the current editor and overview editor. But the benefit is that we do not need to watch for any changes incl positioning of nodes changing, connections changing, nodes being added/deleted and node options changing.

**Comment by tomproberts** [comment]
You probably want to fix the merge conflicts first otherwise it'll reset my review. Or I can fix the conflicts, idm.

**Review by tomproberts — APPROVED** [review]
Awesome stuff! 🎉

**Pull request merged**